library/dialog.js

```diff
@@ -486,7 +488,7 @@ function dlgopen(url, winname, width, height, forceNewWindow, title, opts) {
         ('<div id="%id%" class="modal fade dialogModal" tabindex="-1" role="dialog">%sizeStyle%' +
```

Comment:
@sjpadgett Then how come the JS code involves the CSS modal class... 😂😂😂

Reply:
@tywrenn 
Dialog uses the modal class and more. I wrote dialog mainly because modal is to restrictive when they reside in the iframe. Our viewports keep shrinking so I open dialog in top window where it uses all the viewport of main.php. Also a dialog needs to be able to have callbacks, run functions, custom sizing, render HTML, ajax and nicely return promises on init/shown or close. 

Note the screenshot above where the dialog is outside the batch payment iframe and all the business fields that show in one view then compare a current batch payment modal.(really try it in your local 5.0.2)

Previously we were using a jquery-ui dialog which was pretty nice but we needed to replace, so, instead of bringing in another custom library that will eventually have to be replaced at some point, I wrote a dialog custom to openemr. This is why we also have utility.js.
I'm a firm believer in this concept, if we can write it specific for openemr, we should.

This help ya?

Reply:
oh! plus I hate modals...

Reply:
Hope we're on the same page now but, if not, just ask...

Reply:
This is gonna suck when we have to get rid of frames eventually :(

Reply:
We already got rid of frames. IFrames are here to stay

Reply:
Yeah but they might get rid of those as welllll

Reply:
That's what I like about you @tywrenn, eager.

Reply:
dang, iframes are now bad. When will it end :)